### PR TITLE
add RequestLogging attribute to CreateSiteRuleBody

### PR DESCRIPTION
**Pull request by shawnps**

### CODE REVIEW

File: api_test.go
Comment (stevebartell) [inline api_test.go]: It took a minute for me to realize what is changing for this test request body. To make this clearer, here's a suggestion thats more specific. We just want to know that this SDK is sending and retrieving rules correctly.

```suggestion
	updateSiteRuleBody := createSiteRulesBody
	updateSiteRuleBody.Reason = "a new reason"
```

File: api_test.go
Comment (stevebartell) [inline api_test.go]: There's a case where a site might have a rule of `type:templatedSignal`. These rules don't show in the rules UI.  You see them in the newer templated rules part of the UI. If at least one of those rules are configured on the site then this test case is always going to be wrong - I know only because I have one set up on my personal account and spent way too much time this morning trying to figure out these tests wouldn't pass.

So I suggest just making sure that the `Data` and `TotalCount` lengths are only non-zero.  I might even go as far as suggesting to remove these length assertions only because the point of these tests isn't to exercise the SigSci api, but to make sure this client sdk is hooked up properly.

```suggestion
	if len(readall.Data) != 0 {
		t.Error()
	}

	if readall.TotalCount != 0 {
		t.Error()
	}
```

File: api_test.go
Comment (stevebartell) [inline api_test.go]: Following up to my comment directly above, I'd change this up to just make sure we find the rule in the list
```suggestion
	foundUpdatedRule := false
	for _, r := range readall.Data {
		if r.ID == updateResp.ID {
			foundUpdatedRule = true
		}
	}
	if !foundUpdatedRule {
		t.Error("updated rule not found in GetAllSiteRules")
	}
```

**Review by stevebartell — COMMENTED** [review]
This technically works, but only in a brand new corp.  When I use creds for my personal corp the tests won't pass. So my suggestions are in the sprit of just making sure that the sdk sends and receives data and less about ensuring the sigsci api is behaving properly.

**Review by stevebartell — DISMISSED** [review]

**Comment by stevebartell** [comment]
I just started reviewing https://github.com/signalsciences/go-sigsci/pull/33 and I realize that you're following existing patterns in other tests. So I'm coming back to mark the review as good.  

Overall, these tests depends on a clean corp .. which is a problem to solve another day.

**Comment by shawnps** [comment]
@stevebartell thanks for the review!

> I just started reviewing #33 and I realize that you're following existing patterns in other tests.

Yes exactly, and if we're going to keep these tests they should probably fail right away when it's not a clean corp, and have some setup and teardown functions. 

I think the library should also have some unit tests that don't depend on the network. We can then skip these existing tests if the `SIGSCI_*` environment variables don't exist.

**Comment by shawnps** [comment]
@stevebartell could you take another look? Looks like merging is blocked until I get another review.

File: api.go
Comment (cherylggg) [inline api.go:2153]: is this a public-facing API?

**Review by cherylggg — COMMENTED** [review]
I am here to learn :) let's wait for Steve's review

File: api.go
Comment (shawnps) [reply api.go:2153]: Yup, it's documented [here](https://docs.fastly.com/signalsciences/api/#_corps__corpName__sites__siteName__rules_post).

Comment (cherylggg) [reply api.go:2153]: thanks! with that in mind, do we need to update the doc team to update the API signature?

Comment (shawnps) [reply api.go:2153]: @cherylggg do you mean the Request Example in the docs?  I would say it's not totally necessary to update since the default ("sampled") is documented.

Comment (stevebartell) [reply api.go:2153]: Right, `RequestLogging` should've been added long ago to this SDK, but this repo has sat stale.

**Review by stevebartell — APPROVED** [review]
🚀

File: api.go
Comment (cherylggg) [reply api.go:2153]: thanks for clarifying!

**Review by cherylggg — APPROVED** [review]

**Pull request merged**
